src/Authentication/Authentication/Cmdlets/InvokeGraphRequest.cs

```diff
@@ -0,0 +1,965 @@
+
+        /// <summary>
+        ///     Custom Content Type
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = Constants.UserParameterSet,
+            Position = 11,
+            HelpMessage = "Custom Content Type")]
+        public virtual string ContentType { get; set; }
+
+        /// <summary>
+        ///     Graph Authentication Type
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = Constants.UserParameterSet,
+            Position = 12,
+            HelpMessage = "Graph Authentication Type")]
+        public GraphRequestAuthenticationType Authentication { get; set; }
+
+        /// <summary>
+        /// Gets or sets the SessionVariable property.
+        /// </summary>
+        [Parameter(Position = 13, ParameterSetName = Constants.UserParameterSet,
+            Mandatory = false, HelpMessage = "Session Variable")]
+        [Alias("SV")]
+        public string SessionVariable { get; set; }
+
+        /// <summary>
+        ///  Response Headers Variable
+        /// </summary>
+        [Parameter(Position = 14, ParameterSetName = Constants.UserParameterSet,
+            Mandatory = false,
+            HelpMessage = "Response Headers Variable")]
+        [Alias("RHV")]
+        public string ResponseHeadersVariable { get; set; }
+
+        /// <summary>
+        /// Response Status Code Variable
+        /// </summary>
+        [Parameter(Position = 15, ParameterSetName = Constants.UserParameterSet,
+            Mandatory = false,
+            HelpMessage = "Response Status Code Variable")]
+        public string StatusCodeVariable { get; set; }
+
+        /// <summary>
+        ///     Gets or sets whether to skip checking HTTP status for error codes.
+        /// </summary>
+        [Parameter(Position = 16, ParameterSetName = Constants.UserParameterSet, Mandatory = false, HelpMessage = "Skip Checking Http Errors")]
+        public virtual SwitchParameter SkipHttpErrorCheck { get; set; }
+        /// <summary>
+        /// Gets or sets the Session property.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            Position = 17, ParameterSetName = Constants.UserParameterSet,
+            HelpMessage = "Custom Graph Request Session")]
+        public GraphRequestSession GraphRequestSession { get; set; }
+
+        /// <summary>
+        ///     Custom User Specified User Agent
+        /// </summary>
+        [Parameter(Mandatory = false,
+            Position = 18, ParameterSetName = Constants.UserParameterSet,
+            HelpMessage = "Custom User Specified User Agent")]
+        public string UserAgent { get; set; }
+
+        /// <summary>
+        ///     Wait for .NET debugger to attach
+        /// </summary>
+        [Parameter(Mandatory = false,
+            DontShow = true,
+            HelpMessage = "Wait for .NET debugger to attach")]
+        public SwitchParameter Break { get; set; }
```

Comment:
We shouldn't expose this to customers. This project can already be debugged by pressing F5 and adding breakpoints in Visual Studio. [This](https://docs.poshtools.com/powershell-tools-documentation/chapter1/remote-debugging) is how people generally debug PowerShell scripts.